drivers/SmartThings/zwave-switch/src/qubino-switches/qubino-relays/qubino-flush-2-relay/init.lua

```diff
@@ -11,43 +11,163 @@
+--- @type st.zwave.CommandClass.SensorMultilevel
+local SensorMultilevel = (require "st.zwave.CommandClass.SensorMultilevel")({ version = 7 })
+--- @type st.zwave.constants
+local constants = require "st.zwave.constants"
 
-local QUBINO_FLUSH_2_RELAY_FINGERPRINT = {mfr = 0x0159, prod = 0x0002, model = 0x0051}
+local QUBINO_FLUSH_2_RELAY_FINGERPRINT = { mfr = 0x0159, prod = 0x0002, model = 0x0051 }
 
 local function can_handle_qubino_flush_2_relay(opts, driver, device, ...)
   return device:id_match(QUBINO_FLUSH_2_RELAY_FINGERPRINT.mfr, QUBINO_FLUSH_2_RELAY_FINGERPRINT.prod, QUBINO_FLUSH_2_RELAY_FINGERPRINT.model)
 end
 
 local function component_to_endpoint(device, component_id)
-    if component_id == "main" then
-      return { 1 }
-    elseif component_id == "extraTemperatureSensor" then
-      return { 3 }
-    else
-      local ep_num = math.floor(component_id:match("switch(%d)"))
-      return { ep_num and tonumber(ep_num)}
+  return { 1 }
+end
+
+local function find_child(parent, src_channel)
+  if src_channel == 1 then
+    return parent
+  else
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", src_channel))
+  end
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function get_child_metadata(device, endpoint)
+  local name
+  local profile_name
+  if endpoint ~= 3 then
+    name = string.format("Qubino Switch %d", endpoint)
+    profile_name = "metering-switch"
+  else
+    name = "Qubino Temperature Sensor"
+    profile_name = "child-temperature"
+  end
+
+  return {
+    type = "EDGE_CHILD",
+    label = name,
+    profile = profile_name,
+    parent_device_id = device.id,
+    parent_assigned_child_key = string.format("%02X", endpoint),
+    vendor_provided_label = name
+  }
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    driver:try_create_device(get_child_metadata(device, 2))
+    if device:is_cc_supported(cc.SENSOR_MULTILEVEL) then
+      driver:try_create_device(get_child_metadata(device, 3))
     end
+  end
+end
+
+local function send_refresh_to_endpoint(ep, driver, device)
+  if ep ~= 3 then
+    device:send(SwitchBinary:Get({}, { dst_channels = { ep } }))
+    device:send(Meter:Get({ scale = Meter.scale.electric_meter.WATTS }, { dst_channels = { ep } }))
+    device:send(Meter:Get({ scale = Meter.scale.electric_meter.KILOWATT_HOURS }, { dst_channels = { ep } }))
+  else
+    device:send(SensorMultilevel:Get({ sensor_type = SensorMultilevel.sensor_type.TEMPERATURE }, { dst_channels = { ep } }))
+  end
 end
 
-local function endpoint_to_component(device, ep)
-    if ep == 2 then
-      return string.format("switch%d", ep)
-    elseif ep == 3 then
-      return "extraTemperatureSensor"
+local function do_refresh(driver, device)
+  for i = 1, 2, 1 do
```

Comment:
1 2 1? why twice to ep 1?

Reply:
This is a loop starting from 1, going to 2 and stepping by 1.  In this case I think this would just be clearer to duplicate the call to `send_refersh_to_endpoint` twice.

Reply:
This also currently means that a `refresh` on the child will refresh both enpoint 1 and 2 which doesn't seem correct

Reply:
@varzac fixed